src/nfts/NftsInfoCarousel.tsx

```diff
@@ -0,0 +1,367 @@
+  const { nfts } = route.params
+  const [activeNft, setActiveNft] = useState((nfts && nfts[0]) ?? null)
+  const [isLoading, setIsLoading] = useState(true)
+  const [scaledHeight, setScaledHeight] = useState(360)
+  const { t } = useTranslation()
+
+  const android_ripple = {
+    color: colors.gray2,
+    foreground: true,
+    borderless: true,
+  }
+
+  // Some components that require parent state defined
+  const PlatformSpecificTopBarButtons = () => {
+    return (
+      <View style={styles.topBarButtonsContainerWrapper}>
+        <View style={styles.topBarButtonsContainer}>
+          {Platform.OS === 'ios' && (
+            <TopBarIconButton
+              onPress={() => navigateBack()}
+              icon={<BackChevronStatic />}
+              style={styles.button}
+              testID="NftsInfoCarousel/BackButton"
+            />
+          )}
+          {Platform.OS === 'android' && (
+            <Pressable
+              android_ripple={android_ripple}
+              onPress={() => navigateBack()}
+              style={styles.button}
+              testID="NftsInfoCarousel/BackButton"
+            >
+              <BackChevronStatic />
+            </Pressable>
+          )}
+        </View>
+      </View>
+    )
+  }
+
+  const MainImagePlaceholder = () => {
+    return (
+      <SkeletonPlaceholder
+        borderRadius={8}
+        backgroundColor={colors.gray2}
+        highlightColor={colors.white}
+      >
+        <View
+          style={{
+            height: scaledHeight,
+            width: variables.width,
+            zIndex: -1,
+          }}
+        />
+      </SkeletonPlaceholder>
+    )
+  }
+
+  const NftImageCarousel = () => {
+    return (
+      <View style={styles.nftImageCarouselContainer}>
+        <ScrollView
+          showsHorizontalScrollIndicator={false}
+          horizontal={true}
+          contentContainerStyle={styles.carouselScrollViewContentContainer}
+          style={styles.nftImageCarousel}
+          testID="NftsInfoCarousel/NftImageCarousel"
+        >
+          {/* Only display Nfts with metadata */}
+          {nfts
+            .filter((nft) => nft.metadata)
+            .map((nft) => {
+              return (
+                <View
+                  key={`${nft.contractAddress}-${nft.tokenId}`}
+                  style={styles.nftPreviewImageSharedContainer}
+                  testID={`NftsInfoCarousel/NftThumbnail/${nft.contractAddress}-${nft.tokenId}`}
+                >
+                  <NftThumbnail nft={nft} setActiveNft={setActiveNft} activeNft={activeNft} />
+                </View>
+              )
+            })}
+        </ScrollView>
+      </View>
+    )
+  }
+
+  // Display error screen no Nfts provided
+  if (nfts.length === 0) {
+    return <NftsLoadErrorScreen />
+  }
+
+  return (
+    <SafeAreaView testID="NftsInfoCarousel">
+      <PlatformSpecificTopBarButtons />
+      <ScrollView>
+        {activeNft.metadata && (
```

Comment:
i think the render is a little hard to read because of all the conditions. it seems like we've made the assumption that we want to hide nfts with no metadata in the `NftImageCarousel`, so then we should carry that logic across the whole screen. fwiw i don't really love this and would prefer to show an explicit null metadata state. but if we wanted to have this metadata requirement, i think we should create a new variable that filters out nfts that we don't want to render immediately in `NftsInfoCarousel`, and use this in the render and child components. this means you can remove the sporadic checks and chain operators everywhere

Reply:
I've removed the filtering of Nfts with null metadata. To allow toggling nfts without metadata I had to add some conditional rendering the `NftThumbnail` and a placeholder with a info icon: https://github.com/valora-inc/wallet/pull/3800/commits/7dee7e1bfb5c58e1af48376810814b917175d4e0.